src/react/hooks.js

```diff
@@ -21,3 +27,208 @@ export const useAppEvent = (type, callback) => {
     };
   }, [callback, type]);
 };
+
+const initialParagonThemeState = {
+  isThemeLoaded: false,
+  themeVariant: PARAGON_THEME_VARIANT_LIGHT,
+};
+
+/**
+ * Adds/updates a `<link>` element in the HTML document to load the core application theme CSS.
+ *
+ * @memberof module:React
+ * @param {object} args
+ * @param {string} args.coreThemeUrl The url of the core theme CSS.
+ * @param {string} args.onLoad A callback function called when the core theme CSS is loaded.
+ */
+export const useParagonThemeCore = ({
+  coreThemeUrl,
+  onLoad,
+}) => {
+  useEffect(() => {
+    // If there is no config for the core theme url, do nothing.
+    if (!coreThemeUrl) {
+      return;
+    }
+    let coreThemeLink = document.head.querySelector(`link[href='${coreThemeUrl}']`);
```

Comment:
I haven't tried it, but I think it should work? The code should update the stylesheet links such that the current theme is loaded, so it should respond to changes to runtime config and change of variant as long as there is a mechanism to call this code when such things happen.

Reply:
@arbrandes @xitij2000 Correct, once the MFE runtime config cache expires after the default 5 minutes, it would start pulling in the updated stylesheet links. There is support for changing the theme variant (i.e., light vs. dark) at runtime by exposing a `paragonThemeActions` export from `@edx/frontend-platform/react`.

Example from a consumers POV:

```jsx
import React, { useContext } from 'react';
import { AppContext, paragonThemeActions } from '@edx/frontend-platform/react';

const LoadDarkMode = () => {
  const { paragonTheme } = useContext(AppContext);
  const { setParagonThemeVariant } = paragonThemeActions;
  return (
    <Button onClick={() => paragonTheme.dispatch(setParagonThemeVariant('dark'))}>
      Load dark mode
    </Button>
  );
};
```

Reply:
@arbrandes I also just pushed up a commit to clean up the `<link>` elements for previously used versions should the MFE runtime config API return a different Paragon version specified in the CDN URL (e.g., should it ever get updated periodically without a full page refresh).

Example of switching versions at runtime, with no Flash of Unstyled Content (FoUC): 

https://github.com/openedx/frontend-platform/assets/2828721/d63674cb-fc6f-42eb-9243-69ecb22b957c